### PR TITLE
[Share] CAES-150: Sign and verify requests from client

### PR DESCRIPTION
**Pull request by aburov-4xxi**

### CODE REVIEW

File: src/Event/EventSubscriber/SignatureSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        return [
+            KernelEvents::REQUEST => 'onKernelRequest',
+        ];
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (('false' === getenv('AVAILABLE_REQUEST_SIGNATURE') && 'false' === getenv('STRICT_AUDIT_TRAIL'))
```
Comment (ipopov4xxi) [inline src/Event/EventSubscriber/SignatureSubscriber.php]: Maybe send flags to constructor instead hardcode env variables in code?

File: src/Security/Request/RequestVerifierInterface.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+declare(strict_types=1);
+
+namespace App\Security\Request;
+
+use Symfony\Component\HttpFoundation\Request;
+
+interface RequestVerifierInterface
+{
+    public function verifyRequest(Request $request, string $publicKey): bool;
```
Comment (ipopov4xxi) [inline src/Security/Request/RequestVerifierInterface.php]: If method returns bool, maybe better to name it somthing like `isValidRequest()`?

Comment (aburov-4xxi) [reply src/Security/Request/RequestVerifierInterface.php]: here the meaning is in the verification and not in the validation

File: src/Event/EventSubscriber/SignatureSubscriber.php (hunk 1 above)
Comment (aburov-4xxi) [reply src/Event/EventSubscriber/SignatureSubscriber.php]: What for?

Comment (ipopov4xxi) [reply src/Event/EventSubscriber/SignatureSubscriber.php]: To avoid hardcode environment variables in the code.

Comment (ipopov4xxi) [reply src/Event/EventSubscriber/SignatureSubscriber.php]: Up to you..

**Pull request merged**